### PR TITLE
Adding local executor (driven by docker-compose)

### PR DESCRIPTION
**Pull request by stjohnjohnson**
This executor uses docker-compose on the local machine to run builds (useful for testing).

I didn't implement stream as we're probably going to replace it soon.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    * @param  {Object}   config            A configuration object
+    * @param  {String}   config.buildId    ID for the build
+    * @param  {Response} callback          Callback for when a stream is created
+    */
+    _stream(config, callback) {
+        callback(new Error('Not implemented'));
+    }
+
+    /**
+    * Retreive stats for the executor
```
Comment (d2lam) [inline index.js]: `Retrieve` @nkatzman

Comment (nkatzman) [inline index.js]: i before e when before v

**Comment by FenrirUnbound** [comment]
👍 

Looks good to me

zxcv

**Comment by jer** [comment]
👍 for the name l3l. Not confusing at all.

Also, what happens on build failure here? Does the user just see failures in the output and an exit 0?

**Comment by d2lam** [comment]
👍

**Pull request merged**